python-api/gstswitch/process_monitor.py

```diff
@@ -0,0 +1,157 @@
+"""
+The ProcessMonitor runs a command in a background thread and monitors
+its output- It can block until the command prints a certain string and
+log the full output into a file
+"""
+
+from __future__ import absolute_import, print_function, unicode_literals
+
+import subprocess
+import logging
+import sys
+import os
+import time
+import select
+
+from .exception import ServerProcessError
+from .exception import MatchTimeoutError, MatchEofError, SelectError
+
+
+class ProcessMonitor(subprocess.Popen):
+    """Runs a command in a background-thread and monitors its output
+
+    Can block until the command prints a certain string and log the full
+    output into a file
+    """
+
+    def __init__(self, cmd, cmd_output_target=sys.stderr):
+        self.log = logging.getLogger('server-output-monitor')
+
+        # Logfile to write to
+        self._cmd_output_target = cmd_output_target
+
+        # Internal Buffer to search for matched when wait_for_output is called
+        self._buffer = ""
+
+        self.log.debug("starting subprocess")
+
+        try:
+            super(ProcessMonitor, self).__init__(
+                cmd,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+                bufsize=0,
```

Comment:
It is actually the default and can be removed:
> bufsize, if given, has the same meaning as the corresponding argument to the built-in open() function: 0 means unbuffered, 1 means line buffered, any other positive value means use a buffer of (approximately) that size. A negative bufsize means to use the system default, which usually means fully buffered. The default value for bufsize is 0 (unbuffered).
I think I coyied it from the expect-module or something…

It seems a value of 4096 will not hurt, so I'll raise bufsize to 4096 as suggested in the `subprocess.Popen` docs